src/main/java/com/guciowons/shoppingbasket/Basket/BasketSummarizer.java

```diff
@@ -0,0 +1,35 @@
+package com.guciowons.shoppingbasket.Basket;
+
+import com.guciowons.shoppingbasket.Product.ProductDao;
+import org.springframework.stereotype.Component;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+@Component
+public class BasketSummarizer {
+    public BasketSummarized summarizeBasket(HashMap<Integer, Integer> basketProducts, ProductDao productDao){
+        return new BasketSummarized(
+                productsMapToList(basketProducts, productDao),
+                countPrices(basketProducts, productDao));
+    }
+
+    private BigDecimal countPrices(HashMap<Integer, Integer> basketProducts, ProductDao productDao){
+        List<BasketSummarized.MultiProduct> products = productsMapToList(basketProducts, productDao);
```

Comment:
Ogólnie spoko i to zadziała, ale popatrz sobie znowu na Stream API. Takie przekształcenie można zrobić w jednym łańcuszku bez tworzenia niepotrzebnych zmiennych, coś w stylu:
```
basketProducts.stream()
  .map(...) -- zamień klucz na produkt
  .map(...) -- wylicz wartość produktu, możesz to opakować w jakiś obiekt pomocniczy
  .reduce(...) -- zwiń wszystko w obiekt basket summarized, albo po prostu do sumy cen
```